src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/ExclusionZoneService.kt

```diff
@@ -143,7 +153,14 @@ class ExclusionZoneService(
       .findById(uploadIds.first())
       .orElseThrow { EntityNotFoundException("$conditionId") }
 
-    return upload.fullSizeImage
+    return when {
+      upload.fullSizeImageDsUuid != null ->
+        documentService.getDocument(upload.fullSizeImageDsUuid!!)
+          .also { it.verifyCheckSum(upload.fullSizeImageDsChecksum) }
+
+      upload.fullSizeImage != null -> upload.fullSizeImage
+      else -> null
+    }
```

Comment:
if an image exists in the document store use it, otherwise fallback to DB version. 
The aim is to flush out issues early